sled-agent/src/bin/sled-agent-sim.rs

```diff
@@ -99,9 +113,28 @@ async fn do_run() -> Result<(), CmdError> {
         },
     };
 
+    let tls_certificate = match (args.rss_tls_cert, args.rss_tls_key) {
+        (None, None) => None,
+        (Some(cert_path), Some(key_path)) => {
+            let cert_bytes = std::fs::read(&cert_path)
+                .with_context(|| format!("read {:?}", &cert_path))
+                .map_err(|e| CmdError::Failure(e.to_string()))?;
+            let key_bytes = std::fs::read(&key_path)
+                .with_context(|| format!("read {:?}", &key_path))
+                .map_err(|e| CmdError::Failure(e.to_string()))?;
+            Some(NexusTypes::Certificate { cert: cert_bytes, key: key_bytes })
+        }
+        _ => {
+            return Err(CmdError::Usage(String::from(
+                "--rss-tls-key and --rss-tls-cert must be specified together",
```

Comment:
Could probably do this with `clap` directly but tbh I always have to go look that up 😅